Packs/CommonScripts/Scripts/CommandLineAnalysis/CommandLineAnalysis.py

```diff
@@ -0,0 +1,669 @@
+        r'\bscp\s+-i\s+[a-zA-Z0-9_.-]+\.pem\s+[a-zA-Z0-9_.-]+\s+[a-zA-Z0-9_.-]+:/.*\b',
+        r'\bftp\s+-n\s+[a-zA-Z0-9_.-]+\s+<<END_SCRIPT.*put\s+.*END_SCRIPT\b',
+        r'\bnc\s+[a-zA-Z0-9_.-]+\s+\d+\s+<\s+[a-zA-Z0-9_.-]+\b',
+        r'\baws\s+s3\s+cp\s+[a-zA-Z0-9_.-]+\s+s3://[a-zA-Z0-9_.-]+/.*\b',
+        r'\bgsutil\s+cp\s+[a-zA-Z0-9_.-]+\s+gs://[a-zA-Z0-9_.-]+/.*\b',
+        r'\baz\s+storage\s+blob\s+upload\s+-f\s+[a-zA-Z0-9_.-]+\s+-c\s+[a-zA-Z0-9_.-]+.*\b',
+        r'\bpowershell.*System\.Net\.WebClient.*UploadFile.*https?://[a-zA-Z0-9_.-]+/.*\b',
+        r'\brsync\s+-avz\s+[a-zA-Z0-9_.-]+\s+[a-zA-Z0-9_.-]+:/.*\b'
+    ]
+
+    matches = []
+    for pattern in patterns:
+        matches.extend(re.findall(pattern, command_line, re.IGNORECASE))
+
+    return matches
+
+
+def check_custom_patterns(command_line, custom_patterns=None):
+
+    matches = []
+    for pattern in custom_patterns:
+        matches.extend(re.findall(pattern, command_line, re.IGNORECASE))
+
+    return matches
+
+
+def is_reserved_ip(ip_str):
+    try:
+        ip_obj = ipaddress.ip_address(ip_str)
+        return (
+            ip_obj.is_private
+            or ip_obj.is_loopback
+            or ip_obj.is_reserved
+            or ip_obj.is_multicast
+            or ip_obj.is_link_local
+        )
+    except ValueError:
+        return False
+
+
+def extract_indicators(command_line):
+    try:
+        indicators = demisto.executeCommand("extractIndicators", {"text": command_line})
+        if indicators and isinstance(indicators, list):
+            contents = indicators[0].get('Contents', {})
+
+            # Parse contents if it's a JSON string
+            if isinstance(contents, str):
+                try:
+                    contents = json.loads(contents)
+                except json.JSONDecodeError:
+                    return []
+
+            if isinstance(contents, dict) and "IP" in contents:
+                return [ip for ip in contents["IP"] if not is_reserved_ip(ip)]
+            elif isinstance(contents, dict) and "IP" not in contents:
+                for key in contents:
+                    return list(contents[key])
+    except Exception as e:
+        demisto.debug(f"Failed to extract indicators: {str(e)}")
+    return []
+
+
+def calculate_score(results):
+    # Define weights for base scoring
+    weights = {
+        "mixed_case_powershell": 25,
+        "reversed_command": 25,
+        "powershell_suspicious_patterns": 25,
+        "credential_dumping": 25,
+        "double_encoding": 25,
+        "amsi_techniques": 25,
+        "malicious_commands": 25,
+        "custom_patterns": 25,
+        "data_exfiltration": 15,
+        "lateral_movement": 15,
+        "windows_temp_path": 10,
+        "indicators": 10,
+        "reconnaissance": 10,
+        "base64_encoding": 5,
+        "suspicious_parameters": 5,
+    }
+
+    # Initialize findings and scores for original and decoded
+    findings = {"original": [], "decoded": []}
```

Comment:
This variable needs type annotation:
>Packs/CommonScripts/Scripts/CommandLineAnalysis/CommandLineAnalysis.py:485: error:
>Need type annotation for "findings"  [var-annotated]
>        findings = {"original": [], "decoded": []}

My guess is the is should be:
```python
findings: dict[str, list[Any]] = {"original": [], "decoded": []}
```
But please verify that.